sjb/actions/pull_request_sync.py

```diff
@@ -12,8 +12,67 @@
-_PR_SYNC_ACTION_TEMPLATE = Template("""oct sync remote {{ repository }} --refspec "pull/${{ '{' }}{{ repository | replace('-', '_') | upper }}_PULL_ID}/head" --branch "pull-${{ '{' }}{{ repository | replace('-', '_') | upper }}_PULL_ID}" --merge-into "${{ '{' }}{{ repository | replace('-', '_') | upper }}_TARGET_BRANCH}" """)
+_PULL_NUMBER_PARAMETER_TEMPLATE = Template("""        <hudson.model.StringParameterDefinition>
+          <name>PULL_NUMBER</name>
+          <description>The pull-request in the openshift/origin repository to test. This is compatible with prow.</description>
+          <defaultValue></defaultValue>
+        </hudson.model.StringParameterDefinition>""")
+
+_PULL_REFS_PARAMETER_TEMPLATE = Template("""        <hudson.model.StringParameterDefinition>
+          <name>PULL_REFS</name>
+          <description>The pull-request(s) in the openshift/origin repository to test. This is compatible with prow and can be used for batch testing.</description>
+          <defaultValue></defaultValue>
+        </hudson.model.StringParameterDefinition>""")
+
+_BUILD_ID_PARAMETER_TEMPLATE = Template("""        <hudson.model.StringParameterDefinition>
+          <name>buildId</name>
+          <description>The ID that prow sets on a Jenkins job in order to correlate it with a ProwJob and bubble statuses up to the correct pull request.</description>
+          <defaultValue></defaultValue>
+        </hudson.model.StringParameterDefinition>""")
+
+_PR_SYNC_TITLE_TEMPLATE = Template("SYNC {{ repository | upper }} PULL REQUEST ${{ '{' }}PULL_NUMBER:-}${{ '{' }}{{ repository | replace('-', '_') | upper }}_PULL_ID:-}")
+_PR_SYNC_ACTION_TEMPLATE = Template("""cat << SCRIPT > unravel-pull-refs.py
+#!/usr/bin/env python
+from __future__ import print_function
+
+import sys
+
+# PULL_REFS is expected to be in the form of:
+#
+# base_branch:commit_sha_of_base_branch,pull_request_no:commit_sha_of_pull_request_no,...
+#
+# For example:
+#
+# master:97d901d,4:bcb00a1
+#
+# And for multiple pull requests that have been batched:
+#
+# master:97d901d,4:bcb00a1,6:ddk2tka
+refs = [r.split(':')[0] for r in sys.argv[1].split(',')][1:]
+pull_refs = []
+for ref in refs:
+    if len(ref.strip()) == 0:
+        continue
+    pull_refs.append(ref)
+
+refstring = ''
+for i, pull_num in enumerate(pull_refs):
+    refstring = '{{ '{' }}{{ '}' }}'.format(refstring+pull_num)
+    if len(pull_refs) > i+1:
+        refstring = '{{ '{' }}{{ '}' }},'.format(refstring)
+print('{{ '{' }}{{ '}' }}'.format(refstring))
+SCRIPT
+chmod +x unravel-pull-refs.py
+
+if [[ -n "${PULL_REFS:-}" ]]; then
+  IFS=',' read -ra REFS <<< "$(./unravel-pull-refs.py $PULL_REFS)"
+  for ref in "${{ '{' }}REFS[@]{{ '}' }}"; do
+      oct sync remote {{ repository }} --refspec "pull/$ref/head" --branch "pull-$ref" --merge-into "${{ '{' }}{{ repository | replace('-', '_') | upper }}_TARGET_BRANCH}"
```

Comment:
We need to merge all PRs in the target branch (usually master) because all the tests run on that branch.